hw/ip/dma/data/dma.hjson

```diff
@@ -513,15 +513,15 @@
               desc: "Copy Operation, Simple copy from source to destination."
             }
             { value: "1",
-              name: "SHA256"
+              name: "COPY_SHA256"
               desc: "Perform inline hashing using SHA256."
             }
             { value: "2",
-              name: "SHA384"
+              name: "COPY_SHA384"
               desc: "Perform inline hashing using SHA384."
             }
             { value: "3",
-              name: "SHA512"
+              name: "COPY_SHA512"
               desc: "Perform inline hashing using SHA512."
             }
```

Comment:
I think there is indeed a good argument that this should be regarded as a 'SHA mode' field rather than a single opcode describing the entire operation.

If, as originally proposed, the DMA is extended to include 'Write only', 'Read only' and/or inline encryption, we run the risk of:
- an explosion in the number of enumeration/opcode values to support all permutations, or
- artificially limiting the functionality/modes of operation.

Generally, I prefer to have the hardware configuration separate out these different values, eg.
- 1 bit: enable reading from source
- 1 bit: enable writing to destination
- 2 bits: SHA mode
- n bits: Encryption mode/properties

Presently, for example, we have the slightly bizarre situation that the SHA digest cannot be computed on a block of source data without also writing the data to a destination.

Reply:
I totally agree. Introducing a set of configuration bits that form the overall operating mode makes total sense compared to a linear opcode. However, at this stage of the design, I would not change this, as it requires both, design changes with new error checking and also DV. For the next stepping, when the DMA also supports single-ended transfers (only reading or writing), we can change this for sure.

> Presently, for example, we have the slightly bizarre situation that the SHA digest cannot be computed on a 
block of source data without also writing the data to a destination.

I guess that's just because the DMA was never designed that way from the beginning to support features like initialization (only writing and no reading) or the pure SHA computation (reading but no writing).

Reply:
Certainly. I wasn't proposing that the more flexible/future-proof scheme be implemented at this point.